frontend/ui/otamanager.lua

```diff
@@ -161,94 +192,111 @@ function OTAManager:checkUpdate()
-                            choice1_text = _("Retry"),
-                            choice1_callback = function()
-                                UIManager:show(InfoMessage:new{
-                                    text = _("Downloading may take several minutes…"),
-                                    timeout = 3,
-                                })
-                                -- Clear the installed package, as well as the complete/incomplete update download
-                                os.execute("rm " .. self.installed_package)
-                                os.execute("rm " .. self.updated_package .. "*")
-                                -- And then relaunch zsync in full download mode...
-                                UIManager:scheduleIn(1, function()
-                                    if OTAManager:zsync(true) == 0 then
-                                        UIManager:show(InfoMessage:new{
-                                            text = _("KOReader will be updated on next restart."),
-                                        })
-                                        -- Make it clear that zsync is done
-                                        if self.can_pretty_print then
-                                            os.execute("./fbink -q -y -7 -pm ' '  ' '")
-                                        end
-                                    else
-                                        -- Make it clear that zsync is done
-                                        if self.can_pretty_print then
-                                            os.execute("./fbink -q -y -7 -pm ' '  ' '")
-                                        end
-                                        UIManager:show(ConfirmBox:new{
-                                            text = _("Error updating KOReader. Would you like to delete temporary files?"),
-                                            ok_callback = function()
-                                                os.execute("rm " .. ota_dir .. "ko*")
-                                            end,
-                                        })
-                                    end
-                                end)
-                            end,
-                            choice2_text = _("Abort"),
-                            choice2_callback = function()
-                                os.execute("rm " .. ota_dir .. "ko*")
-                            end,
-                        })
+        if OTAManager:getOTAType() == "link" then
+            UIManager:show(ConfirmBox:new{
+                text = T(
+                    _("Do you want to update?\nInstalled version: %1\nAvailable version: %2"),
+                    local_version,
+                    ota_version
+                ),
+                ok_text = _("Update"),
+                ok_callback = function()
+                    if Device:isSDL() then
```

Comment:
@pazos quoting from https://github.com/koreader/koreader/pull/4620#issuecomment-465299354
> My idea is basically feed these two arguments to Device:update (full url to download and name for the downloaded package) from OTAManager if Device:isAndroid().

And getting that URL is exactly what this PR is about. Android would just need another two or three lines right here.

Reply:
yeah, I know, but I have little time right now to test, so I went for the quick & dirty way.

Reply:
Oh right, sure.